LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/DownlinkBasicsStationMessage.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaTools.LoRaPhysical
+{
+    using System;
+    using LoRaWan;
+
+    /// <summary>
+    /// JSON of a Downlink message for the Basic Station.
+    /// </summary>
+    public class DownlinkBasicsStationMessage
+    {
+        public string DevEui { get; }
+
+        /// This rx delay control the time we need to wait before sending.
+        /// It is typically 0 for class C devices.
+        public ushort LnsRxDelay { get; }
+
+        public ulong Xtime { get; }
+
+        public uint? AntennaPreference { get; }
+
+        public DataRateIndex DataRateRx1 { get; }
+
+        public DataRateIndex DataRateRx2 { get; }
+
+        public Hertz FrequencyRx1 { get; }
+
+        public Hertz FrequencyRx2 { get; }
+
+#pragma warning disable CA1819 // Properties should not return arrays
+        public byte[] Data { get; }
+#pragma warning restore CA1819 // Properties should not return arrays
+
+
+        public StationEui StationEui { get; }
+
+        public DownlinkBasicsStationMessage()
```

Comment:
Why is an empty constructor allowed? Wouldn't that give an invalid message object?

Reply:
Was like this in the previous class, I tried to replace as less as possible given size of the PR. This empty constructor is used by tests

Reply:
Trying to replace it quickly led to ~20 integration tests failing, if that is fine with you, I propose we tackle this as a fturue work item

Reply:
It's used in only one place in tests (`TestDefaultLoRaRequestHandler`), so why not replace there with a call to the normal constructor with `default` for all arguments? It would have the same effect and your tests should pass.